agent/stats/engine.go

```diff
@@ -296,13 +339,108 @@ func (engine *DockerStatsEngine) GetInstanceMetrics() (*ecstcs.MetricsMetadata,
 	return metricsMetadata, taskMetrics, nil
 }
 
+// GetTaskHealthMetrics returns the container health metrics
+func (engine *DockerStatsEngine) GetTaskHealthMetrics() (*ecstcs.HealthMetadata, []*ecstcs.TaskHealth, error) {
+	var taskHealths []*ecstcs.TaskHealth
+	idle := engine.isHealthContainerIdle()
+	metadata := &ecstcs.HealthMetadata{
+		Cluster:           aws.String(engine.cluster),
+		ContainerInstance: aws.String(engine.containerInstanceArn),
+		Fin:               aws.Bool(idle),
+		MessageId:         aws.String(uuid.NewRandom().String()),
+	}
+
+	if idle {
+		return metadata, taskHealths, nil
+	}
+
+	engine.lock.RLock()
+	defer engine.lock.RUnlock()
+
+	for taskARN := range engine.tasksToHealthCheckContainers {
+		taskHealth := engine.getTaskHealthUnsafe(taskARN)
+		if taskHealth == nil {
+			continue
+		}
+		taskHealths = append(taskHealths, taskHealth)
+	}
+
+	if len(taskHealths) == 0 {
+		return nil, nil, EmptyHealthMetricsError
+	}
+
+	return metadata, taskHealths, nil
+}
+
 func (engine *DockerStatsEngine) isIdle() bool {
 	engine.lock.RLock()
 	defer engine.lock.RUnlock()
 
 	return len(engine.tasksToContainers) == 0
 }
 
+func (engine *DockerStatsEngine) isHealthContainerIdle() bool {
```

Comment:
can you rename this to `containerHealthsToMonitor()`? we can then do something like 
```
if ok:=engine.containerHealthsToMonitor(); ok {
 // we have containers whose healths are being monitored
} else {
 // no containers' healths are being monitored
}
```

It reads better than `isHealthContainerIdle`, which doesn't exactly convey the intent of the method.